lxd/instance/instance_utils.go

```diff
@@ -917,6 +917,11 @@ func CreateInternal(s *state.State, args db.InstanceArgs, clearLogDir bool, volu
 	if !args.Snapshot {
 		// Unset expiry date since instances don't expire.
 		args.ExpiryDate = time.Time{}
+
+		// Generate a cloud-init instance-id if not provided.
```

Comment:
Is there a reason we cannot use `volatile.uuid` (the instance's UUID) for this?

Reply:
Also it would be good to add to the comment here explaining what it is used for with cloud-init, and what are the ramifications if it changes.

Reply:
Is there a reason it cannot also be generated (if missig) at startup (like volatile.uuid is)?

Reply:
Makes sense thanks!